phantazm-mob/src/main/java/com/github/phantazmnetwork/mob/config/MobModelConfigProcessor.java

```diff
@@ -0,0 +1,131 @@
+        this.goalCreatorConfigProcessor = Objects.requireNonNull(goalCreatorConfigProcessor,
+                "goalCreatorConfigProcessor");
+        this.itemStackConfigProcessor = Objects.requireNonNull(itemStackConfigProcessor,
+                "itemStackConfigProcessor");
+        this.miniMessage = Objects.requireNonNull(miniMessage, "miniMessage");
+    }
+
+    @Override
+    public @NotNull MobModel dataFromElement(@NotNull ConfigElement element) throws ConfigProcessException {
+        MinestomDescriptor descriptor = descriptorProcessor.dataFromElement(element.getElement("descriptor"));
+
+        ConfigList goalCreatorsGroupsList = element.getListOrThrow("goalCreators");
+        Collection<Iterable<GoalCreator>> goalCreatorsGroups = new ArrayList<>(goalCreatorsGroupsList.size());
+        for (ConfigElement goalCreatorsElement : goalCreatorsGroupsList) {
+            if (!goalCreatorsElement.isList()) {
+                throw new ConfigProcessException("goal creators are not a list");
+            }
+
+            ConfigList goalCreatorsList = goalCreatorsElement.asList();
+            Collection<GoalCreator> goalCreators = new ArrayList<>(goalCreatorsList.size());
+            for (ConfigElement goalCreatorElement : goalCreatorsList) {
+                goalCreators.add(goalCreatorConfigProcessor.dataFromElement(goalCreatorElement));
+            }
+
+            goalCreatorsGroups.add(goalCreators);
+        }
+
+        String displayNameString = element.getStringOrDefault((String) null, "displayName");
+        Component displayName = displayNameString == null ? null : miniMessage.deserialize(displayNameString);
+
+        ConfigNode equipmentNode = element.getNodeOrThrow("equipment");
+        Map<EquipmentSlot, ItemStack> equipment = new HashMap<>(equipmentNode.size());
+        for (Map.Entry<String, ConfigElement> entry : equipmentNode.entrySet()) {
+            EquipmentSlot equipmentSlot = EQUIPMENT_SLOT_MAP.get(entry.getKey().toUpperCase());
+            if (equipmentSlot == null) {
+                throw new ConfigProcessException("unknown equipment slot");
+            }
+
+            if (!entry.getValue().isString()) {
+                throw new ConfigProcessException("equipment value is not string");
+            }
+
+            equipment.put(equipmentSlot, itemStackConfigProcessor.dataFromElement(entry.getValue()));
+        }
+
+        float maxHealth = element.getNumberOrThrow("maxHealth").floatValue();
+
+        return new MobModel(descriptor, goalCreatorsGroups, displayName, equipment, maxHealth);
+    }
+
+    @Override
+    public @NotNull ConfigElement elementFromData(@NotNull MobModel model) throws ConfigProcessException {
+        ConfigElement descriptor = descriptorProcessor.elementFromData(model.getDescriptor());
+
+        ConfigList goalCreatorsGroups = new ArrayConfigList();
+        for (Iterable<GoalCreator> creators : model.getGoalCreatorsGroups()) {
```

Comment:
another instance of Collection<GoalCreator> being preferable to Iterable so you have access to size()